Tests/Service/Serializer/NameConverter/HyphenToUnderscoreNameConverterDecoratorTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIComponentsBundle\Tests\Service\Serializer\NameConverter;
+
+use Auto1\ServiceAPIComponentsBundle\Service\NameConverter\HyphenToUnderscoreNameConverterDecorator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class HyphenToUnderscoreNameConverterDecoratorTest extends TestCase
+{
+    /** @var NameConverterInterface|MockObject */
+    private $decoratedConverterMock;
```

Comment:
@sklabinsky should we use phpspec/prophecy here like we do in the most of ours projects now?

Reply:
@egorkrestinin1 @sklabinsky Just to know, what is the main advantage of using `phpspec/prophecy` instead `MockObject`?

Reply:
Very good question!
We do have an agreement to use **Prophecy**.
@aviator-ua the advantage is to have all tests written in one approach. And the choice was made at some point by simple voting in our php community at that time.
Now we have Prophecy in most of new projects and I'd say - let's keep being consistent.